File: src/main/scala/s/chain_of_responsibility/inheritance/supports/NoSupport.scala
package s.chain_of_responsibility.inheritance.supports

import s.chain_of_responsibility.Trouble
import s.chain_of_responsibility.inheritance.Support

final class NoSupport(name: String, next: Option[Support])
    extends Support(name, next) {
  def this(name: String) = {
    this(name, None)
  }

  override protected def resolve(trouble: Trouble) = false
}

File: src/main/java/j/chain_of_responsibility/inheritance/supports/NoSupport.java
package j.chain_of_responsibility.inheritance.supports;

import j.chain_of_responsibility.Trouble;
import j.chain_of_responsibility.inheritance.Support;
import org.jetbrains.annotations.NotNull;
import org.jetbrains.annotations.Nullable;

public final class NoSupport extends Support {

  public NoSupport(@NotNull String name, @Nullable Support next) {
    super(name, next);
  }

  public NoSupport(@NotNull String name) {
    this(name, null);
  }

  @Override
  protected boolean resolve(Trouble trouble) {
    return false;
  }
}
